### PR TITLE
Fix the e2e --times flag for running tests more than once

### LINKED ISSUE #4555: e2e.go doesn't honor the "--times" flag for tests
The "--times" flag doesn't seem to work when running tests. Flag seems to be ignored and runs the test only once:

go run hack/e2e.go -v -test --test_args="--times=5 --ginkgo.focus=Events.*"

### PR DESCRIPTION
**Pull request by a-robinson**
Fixes #4555.

cc @fabioy

### CODE REVIEW

File: test/e2e/driver.go [hunk 1]
```diff
@@ -75,7 +75,9 @@ func RunE2ETests(authConfig, certDir, host, repoRoot, provider string, orderseed
 		r = append(r, reporters.NewJUnitReporter(path.Join(reportDir, "junit.xml")))
 	}
 	// Run the existing tests with output to console + JUnit for Jenkins
-	ginkgo.RunSpecsWithDefaultAndCustomReporters(&passed, "Kubernetes e2e Suite", r)
+	for i := 0; i < times && passed; i++ {
+		ginkgo.RunSpecsWithDefaultAndCustomReporters(&passed, fmt.Sprintf("Kubernetes e2e Suite run %d of %d", i+1, times), r)
+	}
```
Comment (zmerlynn) [inline test/e2e/driver.go]: Hm. I don't think this would interact well with --report_dir, would it? (It would just overwrite the junit.xml).

Comment (a-robinson) [inline test/e2e/driver.go]: Fair point, it does look like the last one run will win. On the plus side, we stop doing new runs if we encounter an error, meaning that any reports that get overwritten will just be reports of successes.

I'd use a different reporter for each run (junit_%d.xml), but I imagine that existing consumers are dependent on the filename.

Comment (zmerlynn) [inline test/e2e/driver.go]: You're actually safe. Jenkins is currently consuming junit*.xml in anticipation of the line 75 comment above.

Comment (a-robinson) [inline test/e2e/driver.go]: Done, thanks.

**Comment by zmerlynn** [comment]
LGTM

**Pull request merged**